src/main/java/com/blackducksoftware/integration/hub/sonar/component/HubVulnerableComponentGatherer.java

```diff
@@ -92,15 +110,22 @@ private void mapMatchedFilesToComponents(Map<String, Set<VersionBomComponentMode
 
     }
 
-    private void mapMatchedFilesToComponent(Map<String, Set<VersionBomComponentModel>> vulnerableComponentMap, VersionBomComponentModel component) {
-        List<MatchedFilesModel> allMatchedFiles = component.getMatchedFiles();
-        logger.debug(String.format("Found %d files.", allMatchedFiles.size()));
-        for (MatchedFilesModel matchedFile : allMatchedFiles) {
-            String fileName = componentHelper.getFileNameFromComposite(matchedFile.getCompositePathContext());
-            if (!vulnerableComponentMap.containsKey(fileName)) {
-                vulnerableComponentMap.put(fileName, new HashSet<VersionBomComponentModel>());
+    private void mapMatchedFilesToComponent(Map<String, Set<ProjectVersionComponentView>> vulnerableComponentMap, ProjectVersionComponentView component) {
+        List<ComponentMatchedFilesView> matchedFiles = new ArrayList<>();
+        try {
+            matchedFiles = blackDuckService.getAllResponses(component, ProjectVersionComponentView.MATCHED_FILES_LINK_RESPONSE);
```

Comment:
In the future, we should also collect from the several other matched files endpoints.

Reply:
Im not sure there are any other match files endpoints I will have to check on this.